flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkResourceContext.java

```diff
@@ -65,7 +66,9 @@ public KubernetesJobAutoScalerContext getJobAutoScalerContext() {
     }
 
     private KubernetesJobAutoScalerContext createJobAutoScalerContext() {
-        Configuration conf = new Configuration(getObserveConfig());
+        Configuration conf =
+                new KeyMigratingConfiguration(
+                        AutoScalerOptions.LEGACY_CONF_PREFIX, getObserveConfig());
```

Comment:
How about traversing the `getObserveConfig()` and put all keys and values to a new Configuration? We can removing the prefix directly during traversing.

If so, we don't need to introduce the `KeyMigratingConfiguration` class.

Reply:
Semantically, there is no difference. The reason I chose the subclass is that I didn't want to touch any other configs apart from the deprecated options. `Configuration` doesn't allow to access the configuration map directly. One has to extract the configuration as string which will runs through a conversion process.

I just did a quick conversion of the code (see latest commit). I don't prefer this approach but let me know if you find the code more readable.